### ISSUE TITLE
LWT dict has bad data if LWT object is deleted.

### ISSUE THREAD

**Issue by benjeffery** [tskit-dev/tskit#1025]
This works:
```python
>>> lwt = _tskit.LightweightTableCollection()
>>> tskit.TableCollection.fromdict(lwt.asdict())
<tskit.tables.TableCollection object at 0x7f116f75fb50>
```
How ever this doesn't:
```python
>>> lwt_dict = _tskit.LightweightTableCollection().asdict()
>>> tskit.TableCollection.fromdict(lwt_dict)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/benj/projects/tskit/python/tskit/tables.py", line 2353, in fromdict
    tables.individuals.set_columns(**tables_dict["individuals"])
  File "/home/benj/projects/tskit/python/tskit/tables.py", line 558, in set_columns
    self.ll_table.set_columns(
ValueError: Bad offset column encoding
```
this is basically the same as:
```python
>>> lwt = _tskit.LightweightTableCollection()
>>> lwt_dict = lwt.asdict()
>>> del lwt
>>> tskit.TableCollection.fromdict(lwt_dict)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/benj/projects/tskit/python/tskit/tables.py", line 2353, in fromdict
    tables.individuals.set_columns(**tables_dict["individuals"])
  File "/home/benj/projects/tskit/python/tskit/tables.py", line 558, in set_columns
    self.ll_table.set_columns(
ValueError: Bad offset column encoding
```

I think this is because the dict encoding uses `PyArray_SimpleNewFromData` whose documentation says "Create an array wrapper around data pointed to by the given pointer. " Which to me sounds like no copy is taken and it is up to us to make sure the data has the same lifetime as any array object we've made. I guess the right thing to do here is to make a copy - although I find this unsatisfying and do wonder if we should grasp the nettle of dealing with this and `.tables` without a copy. I guess it only matters for very large tree sequences though, unless I've misunderstood?

In summary - everything has worked thus far as the lifetime of our LWTs has always been longer than the dicts they produce.

**Comment by jeromekelleher** [comment]
> I think this is because the dict encoding uses PyArray_SimpleNewFromData whose documentation says "Create an array wrapper around data pointed to by the given pointer. " Which to me sounds like no copy is taken and it is up to us to make sure the data has the same lifetime as any array object we've made.

You're right, this is what's happening.This is a serious bug and we should fix it. The right thing to do is create a new array and copy the data from the table in. The encoding should be independent of the underlying tables, I don't think there's any good reason for not copying here.

We should ideally fix this before the next release, since we're going to be encouraging others to use the lwt interface.
